1.3/spec.md

```diff
@@ -0,0 +1,222 @@
+`tile_row`, and `tile_data`. A typical create statement for the `tiles` table:
+
+    CREATE TABLE tiles (zoom_level integer, tile_column integer, tile_row integer, tile_data blob);
+
+It is common to create an index for this table:
+
+    CREATE UNIQUE INDEX tile_index on tiles (zoom_level, tile_column, tile_row);
+
+#### Content
+
+The tiles table contains tiles and the values used to locate them.
+The `zoom_level`, `tile_column`, and `tile_row` columns follow the
+[Tile Map Service Specification](http://wiki.osgeo.org/wiki/Tile_Map_Service_Specification) in
+their construction, but in a restricted form:
+
+**The [global-mercator](http://wiki.osgeo.org/wiki/Tile_Map_Service_Specification#global-mercator) (aka Spherical Mercator) profile is assumed**
+
+Note that in the TMS tiling scheme, the Y axis is reversed from the coordinate system commonly used in the URLs
+to request individual tiles, so the tile commonly referred to as 11/327/791 is inserted as
+`zoom_level` 11, `tile_column` 327, and `tile_row` 1256, since 1256 is 2^11 - 1 - 791.
+
+The `tile_data blob` column contains raw image or vector tile data in binary.
+
+A subset of file formats are permitted:
+
+* `png`
+* `jpg`
+* `pbf`
+* `mvt`
+
+### Grids
+
+_See the [UTFGrid specification](https://github.com/mapbox/utfgrid-spec) for
+implementation details of grids and interaction metadata itself: the MBTiles
+specification is only concerned with storage._
+
+#### Schema
+
+The database can have optional tables named `grids`, `grid_data`.
+
+The `grids` table must yield four columns named `zoom_level`, `tile_column`,
+`tile_row`, and `grid`. A typical create statement for the `grids` table:
+
+    CREATE TABLE grids (zoom_level integer, tile_column integer, tile_row integer, grid blob);
+
+The `grid_data` table must yield five columns named `zoom_level`, `tile_column`,
+`tile_row`, `key_name`, and `key_json`. A typical create statement for the `grid_data` table:
+
+    CREATE TABLE grid_data (zoom_level integer, tile_column integer, tile_row integer, key_name text, key_json text);
+
+#### Content
+
+The `grids` table contains UTFGrid data, gzip compressed.
+
+The `grid_data` table contains grid key to value mappings, with values encoded
+as JSON objects.
+
+## Vector tileset metadata
+
+As mentioned above, Mapbox Vector Tile tilesets must include a `json` row in the `metadata` table
+to summarize what layers are available in the tiles and what attributes are available for the
+features in those layers.
+
+The row contains the string representation of a JSON object with a `vector_layers` key, whose value is an array of layers.
+Each layer is a JSON object with the following keys:
+
+* `id`: The layer ID, used for a [CartoCSS selector](https://tilemill-project.github.io/tilemill/docs/guides/selectors/) or [Mapbox GL Style layer](https://www.mapbox.com/mapbox-gl-style-spec/#layers).
+* `description`: A human-readable description of the layer's contents (or empty string if not available).
```

Comment:
My understanding is that the implementation requires it to be present even if empty. I should verify that this is actually true.